src/System.Private.CoreLib/shared/System/Resources/ResourceManager.cs

```diff
@@ -566,9 +551,25 @@ protected static Version GetSatelliteContractVersion(Assembly a)
                 throw new ArgumentNullException(nameof(a), SR.ArgumentNull_Assembly);
             }
 
-            // Return null. The calling code will use the assembly version instead to avoid potential type
-            // and library loads caused by CA lookup. NetCF uses the assembly version always.
-            return null;
+            string v = a.GetCustomAttribute<SatelliteContractVersionAttribute>()?.Version;
+            if (v == null)
+            {
+                // Return null. The calling code will use the assembly version instead to avoid potential type
+                // and library loads caused by CA lookup. NetCF uses the assembly version always.
+                return null;
+            }
+
+            if (!Version.TryParse(v, out Version version))
+            {
+                // Note we are prone to hitting infinite loops if mscorlib's
+                // SatelliteContractVersionAttribute contains bogus values.
+                // If this assert fires, please fix the build process for the
+                // BCL directory.
+                Debug.Assert(a == typeof(object).GetTypeInfo().Assembly, System.CoreLib.Name + "'s SatelliteContractVersionAttribute is a malformed version string!");
+                throw new ArgumentException(SR.Format(SR.Arg_InvalidSatelliteContract_Asm_Ver, a.ToString(), v));
+            }
+
+            return version;
```

Comment:
The code for `SatelliteContractVersionAttribute` was not present in CoreCLR. There is issue filed for it - #17816 - and I am not sure whether there was some specific reason to omit it.